cdc/sinkv2/eventsink/mq/dmlproducer/kafka_dml_producer.go

```diff
@@ -70,35 +66,16 @@ func NewKafkaDMLProducer(
 
 	closeCh := make(chan struct{})
 	failpointCh := make(chan error, 1)
-	asyncProducer, err := client.AsyncProducer(changefeedID, closeCh, failpointCh)
+	asyncProducer, err := factory.AsyncProducer(closeCh, failpointCh)
 	if err != nil {
-		// Close the client to prevent the goroutine leak.
```

Comment:
Ditto. The removal of async close logic

Reply:
It's moved to `saramaAsyncProducer.Close`